scopus-import/src/main/java/no/sikt/nva/scopus/conversion/ContributorExtractor.java

```diff
@@ -41,27 +43,27 @@ public class ContributorExtractor {
     private final List<CorrespondenceTp> correspondenceTps;
     private final List<AuthorGroupTp> authorGroupTps;
     private final List<Contributor> contributors;
-    private final PiaConnection piaConnection;
-    private final CristinConnection cristinConnection;
     private final NvaCustomerConnection nvaCustomerConnection;
     private final AffiliationGenerator affiliationGenerator;
+    private final CristinPersonRetriever cristinPersonRetriever;
 
     public ContributorExtractor(List<CorrespondenceTp> correspondenceTps, List<AuthorGroupTp> authorGroupTps,
                                 PiaConnection piaConnection, CristinConnection cristinConnection,
                                 NvaCustomerConnection nvaCustomerConnection) {
         this.correspondenceTps = correspondenceTps;
         this.authorGroupTps = authorGroupTps;
         this.contributors = new ArrayList<>();
-        this.piaConnection = piaConnection;
-        this.cristinConnection = cristinConnection;
         this.nvaCustomerConnection = nvaCustomerConnection;
         this.affiliationGenerator = new AffiliationGenerator(piaConnection, cristinConnection);
+        this.cristinPersonRetriever = new CristinPersonRetriever(cristinConnection, piaConnection);
     }
 
     public List<Contributor> generateContributors() {
         var cristinAffiliationsAuthorgroupsTps = affiliationGenerator.getCorporations(authorGroupTps);
+        var cristinPersons = cristinPersonRetriever.retrieveCristinPersons(authorGroupTps);
         var contributors = cristinAffiliationsAuthorgroupsTps.stream()
-                               .map(this::generateContributorsFromAuthorGroup)
+                               .map(cristinAffiliationsAuthorgroup -> generateContributorsFromAuthorGroup(
```

Comment:
notice the contributors side-effect here. 
I think another PR for getting rid of the side-effect is necessary.

Reply:
I agree. Worth looking closer at later.